warnings: -Wformat

while rebasing on 0.53/master, I have:

```
[17/157] Building CXX object CMakeFiles/sgame-native-dll.dir/src/sgame/sg_cmds.cpp.o
Unvanquished/src/sgame/sg_cmds.cpp:1953:29: warning: format specifies type 'char *' but the argument has type 'Str::StringRef' (aka 'BasicStringRef<char>') [-Wformat]
                                     "%s %s %s", G_NextMapCommand(), Cmd::Escape( arg ).c_str(), Cmd::Escape( reason ).c_str() );
                                      ~~         ^~~~~~~~~~~~~~~~~~
Unvanquished/src/sgame/sg_cmds.cpp:1961:26: warning: format specifies type 'char *' but the argument has type 'Str::StringRef' (aka 'BasicStringRef<char>') [-Wformat]
                                     "%s %s", G_NextMapCommand(), Cmd::Escape( arg ).c_str() );
                                      ~~      ^~~~~~~~~~~~~~~~~~
2 warnings generated.
[23/157] Building CXX object CMakeFiles/sgame-native-dll.dir/src/sgame/sg_main.cpp.o
Unvanquished/src/sgame/sg_main.cpp:1415:44: warning: format specifies type 'char *' but the argument has type 'Str::StringRef' (aka 'BasicStringRef<char>') [-Wformat]
                trap_SendConsoleCommand( va( "%s %s %s", G_NextMapCommand(), Cmd::Escape( g_nextMap.Get() ).c_str(), Cmd::Escape( g_nextMapLayouts.Get() ).c_str() ) );
                                              ~~         ^~~~~~~~~~~~~~~~~~
1 warning generated.
```

Those warnings could be (fixed and) turned to -Werror in CI I think.
I do not think this is linked to #2138 but maybe.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -144,8 +144,6 @@ endif()
 try_flag(WARNINGS           "-Wno-error=switch")
 try_flag(WARNINGS           "-Wno-error=unused-function")
 try_flag(WARNINGS           "-W${WARNMODE}unused-parameter")
-try_flag(WARNINGS           "-Wno-error=deprecated-copy")  # TODO: Remove when move to RmlUi is complete.
-try_flag(WARNINGS           "-Wno-error=format")  # TODO: RmlUi...
 try_flag(WARNINGS           "-Wno-maybe-uninitialized")
 
 try_flag(WARNINGS "/wd26812") # "The enum 'XXX' is unscoped. Prefer 'enum class' over 'enum'"

--- a/src/sgame/sg_cmds.cpp
+++ b/src/sgame/sg_cmds.cpp
@@ -1950,7 +1950,7 @@ static void Cmd_CallVote_f( gentity_t *ent )
 		if ( *reason ) // layout?
 		{
 			Com_sprintf( level.team[ team ].voteString, sizeof( level.team[ team ].voteString ),
-			             "%s %s %s", G_NextMapCommand(), Cmd::Escape( arg ).c_str(), Cmd::Escape( reason ).c_str() );
+			             "%s %s %s", G_NextMapCommand().c_str(), Cmd::Escape(arg).c_str(), Cmd::Escape(reason).c_str());
 			Com_sprintf( level.team[ team ].voteDisplayString,
 			             sizeof( level.team[ team ].voteDisplayString ),
 			             "Change to map '%s' layout '%s'", arg, reason );
@@ -1958,7 +1958,7 @@ static void Cmd_CallVote_f( gentity_t *ent )
 		else
 		{
 			Com_sprintf( level.team[ team ].voteString, sizeof( level.team[ team ].voteString ),
-			             "%s %s", G_NextMapCommand(), Cmd::Escape( arg ).c_str() );
+			             "%s %s", G_NextMapCommand().c_str(), Cmd::Escape(arg).c_str());
 			Com_sprintf( level.team[ team ].voteDisplayString,
 			             sizeof( level.team[ team ].voteDisplayString ),
 			             "Change to map '%s'", arg );

--- a/src/sgame/sg_main.cpp
+++ b/src/sgame/sg_main.cpp
@@ -1412,7 +1412,7 @@ static void ExitLevel()
 	}
 	else if ( G_MapExists( g_nextMap.Get().c_str() ) )
 	{
-		trap_SendConsoleCommand( va( "%s %s %s", G_NextMapCommand(), Cmd::Escape( g_nextMap.Get() ).c_str(), Cmd::Escape( g_nextMapLayouts.Get() ).c_str() ) );
+		trap_SendConsoleCommand( Str::Format( "%s %s %s", G_NextMapCommand(), Cmd::Escape( g_nextMap.Get() ), Cmd::Escape( g_nextMapLayouts.Get() ) ).c_str() );
 	}
 	else if ( G_MapRotationActive() )
 	{

